src/extension.ts

```diff
@@ -3,34 +3,54 @@
 *  Licensed under the MIT License.
 *--------------------------------------------------------------------------------------------*/
 
-import * as languageclient from 'vscode-languageclient';
-import * as logger from './logger';
 import * as path from 'path';
-import * as schemacontributor from './schema-contributor';
 import * as vscode from 'vscode';
-import * as schemaassociationservice from './schema-association-service';
-import TelemetryReporter from 'vscode-extension-telemetry';
-const fs = require('fs');
+import { createTelemetryReporter, callWithTelemetryAndErrorHandling, IActionContext, AzureUserInput, registerUIExtensionVariables } from 'vscode-azureextensionui';
+import * as languageclient from 'vscode-languageclient';
+
 import { activateConfigurePipeline } from './configure/activate';
+import { extensionVariables } from './configure/model/models';
+import * as logger from './logger';
+import * as schemaassociationservice from './schema-association-service';
+import * as schemacontributor from './schema-contributor';
 
-const myExtensionId = 'azure-pipelines';
-const telemetryVersion = generateVersionString(vscode.extensions.getExtension(`ms-azure-devops.${myExtensionId}`));
-const telemetryKey = 'ae672644-d394-497c-8c57-98f6eac32342';
+let perfStats = {
+    loadStartTime: Date.now(),
+    loadEndTime: undefined
+};
 
-let reporter: TelemetryReporter;
+const configurePipelineEnabled: boolean = vscode.workspace.getConfiguration('[azure-pipelines]', null).get('configure') ? true : false;
 
 export async function activate(context: vscode.ExtensionContext) {
     logger.log('Extension has been activated!', 'ExtensionActivated');
```

Comment:
Should we also be logging some info/error via logger?

Reply:
logger is basically doing a console.log, we can start sending the function call errors here. Thoughts ?

Reply:
Makes sense. Idea is basically that if user comes to us and say that configuration is not working, we can ask them to share output log which can help us in quick diagnosis without looking at traces.

Reply:
I have added logger to log all errors being sent out as telemetry events.